nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -381,6 +508,119 @@ Nav2Panel::~Nav2Panel()
 {
 }
 
+void Nav2Panel::loophandler()
+{
+  loop = nr_of_loops->displayText().toStdString();
```

Comment:
This function connects with the text box where the user specify the number of loops. Just in case if the user leaves it empty, then the loop button is not enabled, thus the state remains as before.